client/pages/chat.jsx

```diff
@@ -70,6 +93,15 @@ export default class Chat extends React.Component {
       });
   }
 
+  componentDidUpdate() {
+    // this is helping the chat box scroll to the bottom to view the
+    // most recent message at all times
+    const scroll =
+      this.chatContainer.current.scrollHeight -
+      this.chatContainer.current.clientHeight;
+    this.chatContainer.current.scrollTo(0, scroll);
```

Comment:
It's important to include a condition in your `componentDidUpdate` so that you aren't spamming the DOM on every single re-render.

```js
componentDidMount(prevProps, prevState) {
  if (this.state.messages !== prevState.messages) {
    // there's a new message
  }
}
```

https://reactjs.org/docs/react-component.html#componentdidupdate